### ISSUE TITLE
Sample code fails with Ruby 2.4 and Rails 5.0.2

### ISSUE THREAD

**Issue by nilay** [tradegecko/gecko#69]
`client = Gecko::Client.new('<Application ID>', '<Application Secret>')`
`client.access_token = "<access token>"`
`client.Product.where()`

Raises following error:

```
NoMethodError: undefined method `request' for #<String:0x007fb9f44a1928>
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/gecko-ruby-0.2.2/lib/gecko/record/base_adapter.rb:429:in `block in request'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications.rb:166:in `instrument'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/gecko-ruby-0.2.2/lib/gecko/record/base_adapter.rb:421:in `request'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/gecko-ruby-0.2.2/lib/gecko/record/base_adapter.rb:112:in `where'
	from (irb):25
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/railties-5.0.2/lib/rails/commands/console.rb:65:in `start'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/railties-5.0.2/lib/rails/commands/console_helper.rb:9:in `start'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/railties-5.0.2/lib/rails/commands/commands_tasks.rb:78:in `console'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/railties-5.0.2/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/railties-5.0.2/lib/rails/commands.rb:18:in `<top (required)>'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293:in `require'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293:in `block in require'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:259:in `load_dependency'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293:in `require'
	from /Users/nilayanand/workspace/ClickApps/monh/monh-api/bin/rails:9:in `<top (required)>'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:287:in `load'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:287:in `block in load'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:259:in `load_dependency'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:287:in `load'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/commands/rails.rb:6:in `call'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/command_wrapper.rb:38:in `call'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:191:in `block in serve'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:161:in `fork'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:161:in `serve'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:131:in `block in run'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:125:in `loop'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application.rb:125:in `run'
	from /Users/nilayanand/.rvm/gems/ruby-2.4.0/gems/spring-2.0.1/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /Users/nilayanand/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/nilayanand/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'

```

**Comment by bradleypriest** [comment]
Hi @nilay, it looks like you're trying to pass a string through as the `"<access_token>"` variable when it should be an instance of `OAuth2::AccessToken`.

**Comment by gregblass** [comment]
Why would you program an API this way? Serious question. Most people just give an API key and you're good to go?

**Comment by gregblass** [comment]
Linking to this in your documentation is probably overkill: https://tools.ietf.org/html/draft-ietf-oauth-v2-22

**Comment by bradleypriest** [comment]
Hi Greg, authorization is always about a tradeoff between ergonomics and security.

This is why we've provided Privileged Access Tokens for a limited set of usecases alongside our full OAuth2 flow.


The README has been updated since the original issue to be clearer about how to use the privileged access token flow (see https://github.com/tradegecko/gecko#basic-usage-using-a-privileged-access-token), if you have any suggestions on how we could help improve the messaging, I'd love to see a pull request/suggestions.
